tsl/src/continuous_aggs/refresh.c

```diff
@@ -651,29 +658,23 @@ continuous_agg_refresh_internal(const ContinuousAgg *cagg,
 	if (refresh_window.start >= refresh_window.end)
 	{
 		emit_up_to_date_notice(cagg, callctx);
+		SPI_finish();
```

Comment:
Don't you need a check if `SPI_finish` failed, similar to what you do below?

Reply:
Oops... fixed!